### ISSUE TITLE
IPv6 connexion problem in salt-minion

### ISSUE THREAD

**Issue by valentin2105** [saltstack/salt#32430]
### Description of Issue/Question

When I try Salt-Master/Minion connexion only in IPv6 with Salt 2015.8.x, 
The minion don't use the good IPv6 specified in /etc/salt/minion.
### Setup

My Salt-minion config :

```
master: x:x:a:x::a:2
 ipv6: True
```

The problem, in /var/log/salt/minion :

`[salt.crypt       ][DEBUG   ][5095] Initializing new SAuth for ('/etc/salt/pki/minion', 'a2.nautile.in', 'tcp://[x:x:a:x::a]:2')
`
As you see, the IPv6 was not cutted correctly and look like use the port 2. 

For the master, No problem, it cut correctly the IPv6 :
`tcp://[x:x:a:x::a:2]:4505`
### Steps to Reproduce Issue

Try IPv6 only between Master & Minion on Salt 2015.8.x
### Versions Report

```
Salt Version:
           Salt: 2015.8.8

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.4 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.4 
```

**Comment by jfindlay** [comment]
@valentin2105, thanks for reporting this separate issue.

**Comment by valentin2105** [comment]
You need more details for this Issue ?

**Comment by jfindlay** [comment]
@valentin2105, no your description is good.  I added the label so that we know we need to add an automated test for it.

**Comment by rallytime** [comment]
@valentin2105 This issue was caused by the addition in #25021. This should be fixed now at the HEAD of the 2016.3 branch currently and will be available in the next release of both the 2016.3 and 2016.11 branches. The fixes specifically related to this issue are #38289 and #39766.

This issue is also related to #39336.
